devtools/gradle/gradle-application-plugin/src/main/java/io/quarkus/gradle/tasks/AbstractQuarkusExtension.java

```diff
@@ -75,7 +75,24 @@ private BaseConfig buildBaseConfig() {
         // Using common code to construct the "base config", which is all the configuration (system properties,
         // environment, application.properties/yaml/yml, project properties) that is available in a Gradle task's
         // _configuration phase_.
-        EffectiveConfig effectiveConfig = buildEffectiveConfiguration(Collections.emptyMap());
+        Set<File> resourcesDirs = getSourceSet(project, SourceSet.MAIN_SOURCE_SET_NAME).getResources().getSourceDirectories()
+                .getFiles();
```

Comment:
I'm not sure `MAIN_SOURCE_SET_NAME` always exists? Or did we check that before?

Reply:
This was inlined from https://github.com/quarkusio/quarkus/pull/41337/files#diff-2c6f0d62b0763af665a1bc242ad078ce25794f20632651181549d838e3452998L118. I can revert that back and put it into two methods.